cmd/agent/subcommands/run/command.go

```diff
@@ -501,7 +502,15 @@ func startAgent(
 	// start remote configuration management
 	var configService *remoteconfig.Service
 	if pkgconfig.IsRemoteConfigEnabled(pkgconfig.Datadog) {
-		configService, err = remoteconfig.NewService()
+		apiKey := pkgconfig.Datadog.GetString("api_key")
+		if pkgconfig.Datadog.IsSet("remote_configuration.api_key") {
+			apiKey = pkgconfig.Datadog.GetString("remote_configuration.api_key")
+		}
+		apiKey = configUtils.SanitizeAPIKey(apiKey)
+		baseRawURL := configUtils.GetMainEndpoint(pkgconfig.Datadog, "https://config.", "remote_configuration.rc_dd_url")
+		traceAgentEnv := configUtils.GetTraceAgentDefaultEnv(pkgconfig.Datadog)
+
+		configService, err = remoteconfig.NewService(pkgconfig.Datadog, apiKey, baseRawURL, hostnameDetected, remoteconfig.WithTraceAgentEnv(traceAgentEnv))
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I think this logic should rather be inside remote config code, especially since it takes the config anyway, you could remove the `apiKey`, `baseRawURL`, and `hostnameDetected` arguments entirely (and compute them at the start of `NewService`, or at worst in an helper function

Reply:
Can you create a function to factorize the logic of this code with the one in `cmd/cluster-agent/subcommands/start/command.go`

Reply:
@pgimalac This logic is not inside the remote config code because hostnameDetected requires the package `pkg/util/hostname`, and baseRawURL requires `pkg/config/utils`, which both aren't Go modules :-/
I can move apiKey to the remote config code though

Reply:
I moved the logic to cmd/agent/common